### ISSUE TITLE
Mulberry-Symbols - further developements and translation

### ISSUE THREAD

**Issue by LarsRM** [mulberrysymbols/mulberry-symbols#41]
For a project for nonverbal people with aphasia we are looking for appropriate symbols for adults and found the Mulberry symbols via Internet search. We like the symbols (especially for this user group) and we would like to know if there are chances for further developements (more symbols and country related symbols) and translation into other languages (German in our case)?

**Comment by SteveALee** [comment]
Thanks @LarsRM it's great to hear from you.

We are now exploring how to support new symbols. This involves 

* Finding designers - the previous one is interested
* Finding people to colour and translate
* Figuring out the open process for request and review
* Coding any tools required for the process

You might also like to comment on a couple of other issues we have: 
*  #32 new symbols requests
*  #36 how wonuld you like to use the symbols.

**Comment by neilt1700** [comment]
Just wondering if there is a plan for developing new symbols. I would be keen to be involved if there is something to fit into.

I was having a conversation with a teacher who was previously an illustrator, and we were wondering whether there might be any scope for getting students on illustration degrees involved by having the creation of a quantity of symbols as a degree project.

**Comment by shayc** [comment]
@neilt1700 I'm not sure if there are any plans on extending Mulberry Symbols (@SteveALee can better answer this - I'm not a maintainer).

Do you know about [Global Symbols](https://globalsymbols.com/)?

**Comment by SteveALee** [comment]
@neilt1700 (thanks @shayc) 

Thanks for asking! Yes, there is a general desire to expand the symbol set but no concrete plans yet.

There has been discussion with a commercial user of the symbols wanting some specific new symbols. The original designer has also expressed interest in providing more symbols. This has not evolved past the talk stage though.

I *love* the idea of student contribution as part of the their project and would happily support that with my time here.  I've been involved is several student accessibility competitions (eg Project:Possibility, AAATE and ICCHP) and love the energy students bring and that open devleopment can be of value on their CVs.

We'd need to sort out style guidelines and a symbol review process (as supported by the original straight-street home of the symbols) . May some sort of sponsorship or other funding could be found.

**Comment by eadraffan** [comment]
Dear All

 

I totally support Steve in this endeavour, as we desperately need a free and open licenced symbol set that is more suitable for adults.  It would be wonderful if this initiative happen. 

 

I never normally advertise on a group such as this, but please may I mention that we have system available for voting on AAC symbol acceptance  https://globalsymbols.com/ and we are very happy to help with the process. We have been supporting a UNICEF project and worked in a similar way to develop the Tawasol symbol set.  I have attached a paper about the way it works,  that was also presented at AAATE in 2017. 

 

Best wishes 

E.A.

**Comment by SteveALee** [comment]
Thanks EA - I'd say that's more an offer of help than an advertisement!

I hadn't realised the voting system was ready for use now. Brilliant and perfect!

**Comment by neilt1700** [comment]
(1) Style Guide
(2) Words List
(3) Funding

In order to make symbols development easier:

- Has the original style guide been found? Or can a new one be created?

- Is there a target words list for creating symbols?

To look at word coverage for symbol sets I have been comparing the words in these sets to:
- the top 5000 words in the Corpus of Contemporary American English (COCA) - plus a few extra with UK spelling versions of some of these words.
- a set of first 100 words (from Cambridgshire SLT department as it happens).
- I would also want to include a core words list - here's an example: core words only from https://thinksmartbox.com/product/super-core-new/ I have a set which I am using and adapting in a special school.

Our symboliser (and I guess other search tools) looks for symbols based on their filenames. To make this more efficient I have created versions of symbol sets (ARASAAC and Mulberry) where words that don't add anything much to the meaning are removed, such as "to" in "to come", "to go", "to sit". Or, where more than one word is needed to define the meaning, link them together with a hyphen, e.g. "how-are-you", "horse-riding". We also use the Porter Stemmer to bring up more symbols (compare the stem of what the user enters to the stems of words on the symbol names, e.g. "receive", "receives", "received" all stem to "receiv").

I've also used a thesaurus to add more words to each individual symbol - this significantly increases coverage.

I've created a macro in excel which lets you add words to symbol names efficiently as well as computing the set's coverage of words in the lists above. I am happy to share this.

- I guess to really get the project going it would need some funding

I can see that some symbols could be made by volunteers, or students on illustration courses. However I am guessing that this might not produce enough symbols. I guess the set would need something like another 5000 symbols to make it really useful - in my experience, at present there aren't enough words in it to make it useful enough in a lot of situations (e.g. creating communication books for children and adults, adding symbols to documents such as timetables or minutes for people with learning disabilities).

Are there are any thoughts on this? Is Mulberry Symbols set up as a charity to facilitate funding applications? I see Straight Street was registered as a charity. As a Community Interest Company, we have had a small amount of experience in fundraising.

**Comment by SteveALee** [comment]
@neilt1700 This is a quick placeholder repsonse to say thank you and I'll reply ASAP. but for now the "easy" one

> Are there are any thoughts on this [funding]? Is Mulberry Symbols set up as a charity to facilitate funding applications? I see Straight Street was registered as a charity. As a Community Interest Company, we have had a small amount of experience in fundraising.

My initial purpose was to save the symbols after the Straight-Street charity folded and Garry Paxton move don. I would love to encourage / support growth but was waiting to see what interest we got. It seems there is some. I work for myself as OpenDirective Ltd but did explore making it a CIC or maybe start a new one just for the symbols so we coud attract more funding options.

Anyway it's usually seen as good Open Source practice to keep the funding aspect spearate from the daily project discussion: I'm happy to tak it offline :)

**Comment by SteveALee** [comment]
@neilt1700 

>   Has the original style guide been found? Or can a new one be created?

I just asked again. We might need to creat a new one. Colours can be worked out but other features might be implicit.

*    Is there a target words list for creating symbols?

Not yet. The way it worked before was each iteration added symbols based on a requests. I agree this would a be a good thing to work on and I expect @eadraffan might have something from Global Symbols which is working on cross referencing and concept coding 

> To look at word coverage for symbol sets I have been comparing the words in these sets to:
[clip]

this sounds like useful work indeed 

> Our symboliser (and I guess other search tools) looks for symbols based on their filenames. To make this more efficient I have created versions of symbol sets (ARASAAC and Mulberry) where words that don't add anything much to the meaning are removed, such as "to" in "to come", "to go", "to sit". Or, where more than one word is needed to define the meaning, link them together with a hyphen, e.g. "how-are-you", "horse-riding". We also use the Porter Stemmer to bring up more symbols (compare the stem of what the user enters to the stems of words on the symbol names, e.g. "receive", "receives", "received" all stem to "receiv").

There's an aweful lot to concider here in how we name, map and reference symbols. Again Global Symbols is an attempt to provide a canonical ID system with mapping between equivalents in different sets and internatinalisation and localisation. I think it builds on the Concept Coding framework. Some projects  I worked on use filesnames as you say and for StraighStreet we looked at i18n though localising filenames and mamping them in a db. When talking to Barney Hawes I think Grid uses an ID and mapping system. For the W3C WAI personalisation use of symbols Bliss symbols are currently used as the ID space with a way to map to a specific symbol set. There was talk of using Unicode too.

In mulberry the 'to' convention is for verbs and does add something when a word can be a verb or noun and thus there are 2 different symbols for the same word. Homonyms, synonyms etc all add fun to symbol identification.
 
> I've also used a thesaurus to add more words to each individual symbol - this significantly increases coverage.
>I've created a macro in excel which lets you add words to symbol names efficiently as well as computing the set's coverage of words in the lists above. I am happy to share this.

It does sound useful.
We looked at adding tags to symbols but that gets a littie hard to manage.

In summary then, there's a lot involved in symbol naming and maping and we should build on the work of others like CCF, Global Symbols, W3C and more. I prefer that than coming up with yet another variant for mulberry. THe problem is all the work is still quite immature and a the problem space large and complex.

**Comment by neilt1700** [comment]
> There are several word lists that can be used but the one that was researched for older students came from The Dynamic Learning Maps Core Vocabulary (Chapel Hill) https://www.med.unc.edu/ahs/clds/resources/core-vocabulary/
That's a really useful link @eadraffan.

Symbols naming
I guess for automatic symbolising of text having grammatical markers is useful. I guess for trying to make websites more accessible that might be something. However to really make a chunk of text accessible it's much more preferable to select key words to symbolise which support the meaning - symbolising everything can mean you end up with a whole mass of symbols which isn't usually helpful. The important point is that to do this properly there is always going to be someone there doing it who understands the semantics and what is the key information for the intended user group. They might click through some alternative options until they get to the appropriate symbol (e.g. parts of speech, homonyms). 

Also, if an end user wants to add some localised symbols/pictures, I would think that it shouldn't be any more complicated than them naming those images with the word(s) they want to refer to them by.

For flexible use of the symbols, would a combination of having the actual words in the symbol name followed by an id work? The id could then be used to reference any further data  such as pos, a ccf concept and/or translations. Our software ignores anything after an underscore in a file name: from the observation that Arasaac, Sclera and Clarity (Liberator) append an underscore to the filenames of their images followed by a number where there are alternative images for the same word (Clarity also has pos markers here - e.g. _V, _A). 

By the way, I've been looking at the Concept Coding Framework. I can't see how it's actually storing meaning information (I've looked through its website and downloaded it). Is it perhaps using Bliss symbols as a base?

**Comment by SteveALee** [comment]
We could delegate to Global Symbols - see #30
